File: frontend/src/types/expense.ts
export interface Expense {
  id: number;
  event_id: number;
  category: string;
  description?: string;
  amount: number;
  created_at: string;
}

export interface ExpenseCreate {
  event_id: number;
  category: string;
  description?: string;
  amount: number;
}

export interface BudgetSummary {
  event_id: number;
  total_budget: number;
  total_expenses: number;
  remaining_budget: number;
  utilization_percent: number;
  warning?: string;
  expenses_by_category: Record<string, number>;
  expenses: Expense[];
}

File: backend/app/schemas/expense.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List

class ExpenseCreate(BaseModel):
    event_id: int
    category: str
    description: Optional[str] = None
    amount: float

class ExpenseResponse(BaseModel):
    id: int
    event_id: int
    category: str
    description: Optional[str] = None
    amount: float
    created_at: datetime

    class Config:
        from_attributes = True

class BudgetSummary(BaseModel):
    event_id: int
    total_budget: float
    total_expenses: float
    remaining_budget: float
    utilization_percent: float
    warning: Optional[str] = None
    expenses_by_category: dict
    expenses: List[ExpenseResponse]